earth_enterprise/src/common/mutex.cpp

```diff
@@ -0,0 +1,67 @@
+// Copyright 2018 The Open GEE Contributors
```

Comment:
There are some existing mutex classes in khThread.cpp/.h. I think khLockGuard would do the same thing as this class. It might be simpler to use the existing code.

Reply:
:+1:

I see that being done in #708, which, however, also includes some `pip` changes.  We just need a clean PR with the mutex.

Reply:
Any updates w/r these comments?